cranelift-codegen/meta/src/isa/x86/encodings.rs

```diff
@@ -2174,12 +2176,14 @@ fn define_entity_ref(
         is_pic,
     );
 
-    // Stack addresses.
-    //
-    // TODO: Add encoding rules for stack_load and stack_store, so that they
-    // don't get legalized to stack_addr + load/store.
-    e.enc32(stack_addr.bind(I32), rec_spaddr4_id.opcodes(&LEA));
-    e.enc64(stack_addr.bind(I64), rec_spaddr8_id.opcodes(&LEA).rex().w());
+    // Stack accesses.
+    e.enc32(stack_addr.bind(I32), rec_spaddr_ld_id.opcodes(&LEA));
```

Comment:
Is it possible to use the 8-bytes displacement on x86 32 bits, though? It seems like a possible regression, and one that's hard to test, since we only test that the generated machine code matches our encodings.

(One possible way to resolve this would be to just remove this line, in light of the plan to have a new isel/codegen mechanism, since x86 32 bits is badly supported anyways.)

Reply:
I can add the 8bit displacement back.

Reply:
I'm not sure if @bnjbvr meant 8-byte or 8-bit displacement, but to be clear, x86 does not support 8-byte displacements.

I'm suspect of the `spaddr4` encoding for x86 32-bit that was present since while it used `modrm_sib_disp8(out_reg0, sink);` to prepare for an 8-bit displacement, it proceeded to write four bytes of displacement out with  `sink.put4(sp.offset.checked_add(imm).unwrap() as u32);`

I think adding keeping an 8-bit displacement form would be good though, as it would be useful for functions with smaller stack regions.

Reply:
Yeah, that put4 was the reason I removed it.